api/database_cluster_backup.go

```diff
@@ -67,11 +74,34 @@ func (e *EverestServer) CreateDatabaseClusterBackup(ctx echo.Context, namespace
 }
 
 // DeleteDatabaseClusterBackup deletes the specified cluster backup on the specified kubernetes cluster.
-func (e *EverestServer) DeleteDatabaseClusterBackup(ctx echo.Context, namespace, name string) error {
+func (e *EverestServer) DeleteDatabaseClusterBackup(
+	ctx echo.Context,
+	namespace, name string,
+	params DeleteDatabaseClusterBackupParams,
+) error {
+	cleanupStorage := pointer.Get(params.CleanupBackupStorage)
+	// If cleanup is required, we add a finalizer to the backup object.
+	if cleanupStorage {
+		reqCtx := ctx.Request().Context()
+		backup, err := e.kubeClient.GetDatabaseClusterBackup(reqCtx, namespace, name)
+		if err != nil {
+			return err
```

Comment:
Shall we return a more user friendly error?

Reply:
Updated